libgalois/src/analytics/subgraph_extraction/subgraph_extraction.cpp

```diff
@@ -0,0 +1,128 @@
+ * UNIVERSITY EXPRESSLY DISCLAIMS ANY AND ALL WARRANTIES CONCERNING THIS
+ * SOFTWARE AND DOCUMENTATION, INCLUDING ANY WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR ANY PARTICULAR PURPOSE, NON-INFRINGEMENT AND WARRANTIES OF
+ * PERFORMANCE, AND ANY WARRANTY THAT MIGHT OTHERWISE ARISE FROM COURSE OF
+ * DEALING OR USAGE OF TRADE.  NO WARRANTY IS EITHER EXPRESS OR IMPLIED WITH
+ * RESPECT TO THE USE OF THE SOFTWARE OR DOCUMENTATION. Under no circumstances
+ * shall University be liable for incidental, special, indirect, direct or
+ * consequential damages or loss of profits, interruption of business, or
+ * related expenses which may arise from use of Software or Documentation,
+ * including but not limited to those resulting from defects in Software and/or
+ * Documentation, or loss or inaccuracy of data of any kind.
+ */
+
+#include "katana/analytics/subgraph_extraction/subgraph_extraction.h"
+
+#include "katana/TypedPropertyGraph.h"
+#include "katana/analytics/Utils.h"
+
+using namespace katana::analytics;
+
+namespace {
+using edge_iterator = boost::counting_iterator<uint64_t>;
+
+katana::Result<std::unique_ptr<katana::PropertyGraph>>
+SubGraphNodeSet(
+    katana::PropertyGraph* graph, const std::vector<uint32_t>& node_set) {
+  auto subgraph = std::make_unique<katana::PropertyGraph>();
+  if (node_set.empty()) {
+    return std::unique_ptr<katana::PropertyGraph>(std::move(subgraph));
+  }
+
+  uint64_t num_nodes = node_set.size();
+  // Subgraph topology : out indices
+  auto out_indices = std::make_unique<katana::LargeArray<uint64_t>>();
+  out_indices->allocateInterleaved(num_nodes);
+
+  katana::gstl::Vector<katana::gstl::Vector<uint32_t>> subgraph_edges;
+  subgraph_edges.resize(num_nodes);
+
+  katana::GAccumulator<uint64_t> accum_num_edges;
+  katana::do_all(
+      katana::iterate(node_set),
+      [&](const uint32_t& n) {
+        for (auto dest : node_set) {
+          auto edge_range = graph->topology().edge_range(n);
+          auto edge_matched = std::lower_bound(
+              edge_iterator(edge_range.first), edge_iterator(edge_range.second),
+              dest, [=](edge_iterator e, uint32_t n) {
+                return graph->topology().edge_dest(*e) < n;
+              });
+          if (*edge_matched != edge_range.second) {
+            while (*edge_matched == dest) {
+              subgraph_edges[n].push_back(dest);
+              edge_matched++;
+              accum_num_edges += 1;
+            }
+          }
+          (*out_indices)[n] = subgraph_edges[n].size();
+        }
+      },
+      katana::steal(), katana::no_stats(),
+      katana::loopname("SubgraphExtraction"));
+
+  uint64_t num_edges = accum_num_edges.reduce();
+  // Subgraph topology : out dests
+  auto out_dests = std::make_unique<katana::LargeArray<uint32_t>>();
+  out_dests->allocateInterleaved(num_edges);
+
+  // Prefix sum
+  for (uint64_t i = 1; i < num_nodes; ++i) {
+    (*out_indices)[i] += (*out_indices)[i - 1];
```

Comment:
You don't need `accum_num_edge`. `num_edges` should be `out_indices[num_nodes - 1]`